src/SingleArticle.jsx

```diff
@@ -0,0 +1,22 @@
+import { useParams } from 'react-router-dom'
+import { useEffect,useState } from 'react'
+import axios from 'axios'
+import { ArticleCards } from './ArticleCards'
+
+export const SingleArticle=()=>{
+    const [thisArticle,setThisArticle]=useState({})
+    const [thisArticleComments,setThisArticleComments]=useState([])
+    const {article_id}=useParams()
+    useEffect(()=>{
+        axios.get(`https://news-api-p73k.onrender.com/api/articles/${article_id}`)
+            .then(res=>setThisArticle(res.data.article))
+            .then(()=>axios.get(`https://news-api-p73k.onrender.com/api/articles/${article_id}/comments`))
+            .then(res=>setThisArticleComments(res.data.comments))
+            .catch(err=>console.log(err))
+    },[])
+    return (
+        <>
+            {Object.keys(thisArticle).length?(<ArticleCards articles={[thisArticle]} showBody={true}/>):'Loading...'}
```

Comment:
If the user uses the URL to navigate to another article from here, they won't see the 'Loading...' message because the ternary expression will still resolve to true while the other article is fetched. I'd recommend changing this to instead rely on some good old fashioned loading state

Reply:
doesn't useState reset to whatever the default value is whenever they search a new url?